demo-apps/sc-dtls-example-client/src/main/java/org/eclipse/californium/scandium/examples/ExampleDTLSClient.java

```diff
@@ -136,55 +148,87 @@ private void startTest(InetSocketAddress peer) {
 	}
 
 	private void stop() {
-		if (dtlsConnector.isRunning())
+		if (dtlsConnector.isRunning()) {
 			dtlsConnector.destroy();
+		}
 	}
 
 	public static void main(String[] args) throws InterruptedException {
-		int max = 1;
+		int clients = 1;
+		int messages = 100;
 		if (0 < args.length) {
-			max = Integer.parseInt(args[0]);
+			clients = Integer.parseInt(args[0]);
+			if (1 < args.length) {
+				messages = Integer.parseInt(args[1]);
+			}
 		}
-		latch = new CountDownLatch(max);
-		List<ExampleDTLSClient> clients = new ArrayList<>(max);
+		maxMessages = (messages * clients);
+		messageCounter = new CountDownLatch(maxMessages);
+		List<ExampleDTLSClient> clientList = new ArrayList<>(clients);
+		ExecutorService executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors(),
+				new DaemonThreadFactory("Aux#"));
+
+		System.out.println("Create " + clients + " DTLS example clients.");
+
+		final CountDownLatch start = new CountDownLatch(clients);
 
 		// Create & start clients
-		for (int index = 0; index < max; ++index) {
-			ExampleDTLSClient client = new ExampleDTLSClient();
-			client.start();
-			clients.add(client);
+		for (int index = 0; index < clients; ++index) {
+			final ExampleDTLSClient client = new ExampleDTLSClient();
+			clientList.add(client);
+			executor.execute(new Runnable() {
+
+				@Override
+				public void run() {
+					client.start();
+					start.countDown();
+				}
+			});
 		}
+		start.await();
+		System.out.println(clients + " DTLS example clients started.");
 
 		// Get peer address
 		InetSocketAddress peer;
-		if (args.length == 3) {
-			peer = new InetSocketAddress(args[1], Integer.parseInt(args[2]));
+		if (args.length == 4) {
+			peer = new InetSocketAddress(args[2], Integer.parseInt(args[3]));
 		} else {
 			peer = new InetSocketAddress(InetAddress.getLoopbackAddress(), DEFAULT_PORT);
 		}
 
 		// Start Test
 		long nanos = System.nanoTime();
-		for (ExampleDTLSClient client : clients) {
+		long lastMessageCountDown = messageCounter.getCount();
+
+		for (ExampleDTLSClient client : clientList) {
 			client.startTest(peer);
 		}
 
-		// Wait for it
-		latch.await(max * 50, TimeUnit.MILLISECONDS);
+		// Wait with timeout or all messages send.
+		while (!messageCounter.await(DEFAULT_TIMEOUT_NANOS, TimeUnit.NANOSECONDS)) {
+			long current = messageCounter.getCount();
+			if (lastMessageCountDown == current && current < maxMessages) {
+				// no new messages, clients are stale
+				// adjust start time with timeout
+				nanos += DEFAULT_TIMEOUT_NANOS; 
+				break;
+			}
+			lastMessageCountDown = current;
+		}
```

Comment:
It's clearly better way to guess the end of the test.
About statistics, it's better too, but in my tests when I have message lost I still have lower number of message by second.
I suspect that even with this attempt to fix statistic this is no so accurate. (I suppose this is because some clients can stop really soon after start and so less client in parallel means less messages exchanged)